File: linked_lists/swap_nodes_in_pairs.java
class ListNode
{
    int val;
    ListNode next;
    ListNode(int val)
    {
        this.val = val;
    }
}

public ListNode swapPairs(ListNode head)
{
    if (head == null || head.next == null)
    {
        return head;
    }

    ListNode dummy = head.next;
    ListNode prev  = null;

    while (head != null && head.next != null)
    {
        if (prev != null)
        {
            prev.next = head.next;
        }

        prev = head;

        ListNode nextNode = head.next.next;

        head.next.next = head;

        head.next = nextNode;

        head = nextNode;
    }

    return dummy;
}

File: linked_lists/swap_nodes_in_pairs.py
class ListNode:
    def __init__(self, val):
        self.val = val
        self.next = None

def swapPairs(head: ListNode) -> ListNode:
    if not head or not head.next:
        return head
    
    dummy = head.next
    prev = None

    while head and head.next:
        if prev:
            prev.next = head.next

        prev = head

        next_node = head.next.next
        head.next.next = head

        head.next = next_node
        head = next_node
    
    return dummy